Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -970,70 +990,88 @@ KeyCode MapKey (WindowsConsole.ConsoleKeyInfoEx keyInfoEx)
 		case ConsoleKey.Oem7:
 		case ConsoleKey.Oem8:
 		case ConsoleKey.Oem102:
-			//var ret = ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)keyInfo.KeyChar));
-			//if (ret.HasFlag (KeyCode.ShiftMask)) {
-			//	ret &= ~KeyCode.ShiftMask;
-			//}
-			//return ret;
-
 		case ConsoleKey.OemPeriod:
 		case ConsoleKey.OemComma:
 		case ConsoleKey.OemPlus:
 		case ConsoleKey.OemMinus:
-			return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)keyInfo.KeyChar));
+			if (keyInfo.KeyChar == 0) {
+				var mappedChar = MapVKtoChar ((VK)keyInfo.Key);
+				if (mappedChar != 0) {
+					var charVal = (char)(mappedChar & 0x0000FFFF);
+					// An un-shifted character value in the low order word of the return value.
+					// Dead keys (diacritics) are indicated by setting the top bit of the return value. 
+					if ((mappedChar & 0x80000000) != 0) {
+						// Dead key (e.g. Oem2 '~'/'^' on POR keyboard)
+						// Option 1: Throw it out. 
+						//    - Apps will never see the dead keys
+						//    - If user presses a key that can be combined with the dead key ('a'), the right thing happens (app will see '�').
+						//      - NOTE: With Dead Keys, KeyDown != KeyUp. The KeyUp event will have just the base char ('a').
+						//    - If user presses dead key again, the right thing happens (app will see `~~`)
+						//    - This is what Notepad etc... appear to do
+						// Option 2: Expand the API to indicate the KeyCode is a dead key
+						//    - Enables apps to do their own dead key processing
+						//    - Adds complexity; no dev has asked for this (yet).
+						// We choose Option 1 for now.
+						return KeyCode.Null;
+					}
+
+					if (keyInfo.Modifiers.HasFlag (ConsoleModifiers.Control)) {
+						// We get here if Windows detected a Ctrl-Deadkey (like Oem3 '`'/'~` on ENG).
+						// Sadly, the charVal is just the deadkey and subsequent key events do not contain
+						// any info that the previous event was a deadkey.
+						// TODO: We *could* set a global with charVal and then, on the next KeyDown combine them,
+						// but that's super-complex, and WT does not support this either. 
+						// This means Ctrl-DeadKey input is not supported in Terminal.Gui apps
+						return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)charVal);
+					}
+					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)charVal);
+				} else {
+					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)keyInfo.Key);
+				}
+			} else {
+				// Strip off Shift - We got here because they KeyChar from Windows is the shifted char (e.g. "�") and passing on Shift
+				// would be redundant.
+				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers & ~ConsoleModifiers.Shift, (KeyCode)keyInfo.KeyChar );
```

Comment:
I think we must be consistent with the using the only shift key, as it is used on A..Z. Because of this behavior using only the shift key alone was never added to the `KeyCode` but only to the `ModifiersKeys`. Now that the A..Z  is adding and removing the shift key before send through a `Key` event, I think we should follow the same principle, as Windows does.